spirv_msl.hpp

```diff
@@ -787,6 +792,8 @@ class CompilerMSL : public CompilerGLSL
 	uint32_t view_mask_buffer_id = 0;
 	uint32_t dynamic_offsets_buffer_id = 0;
 
+	bool does_shader_write_sample_mask = false;
```

Comment:
The API user shouldn't have to declare if the shader writes the sample mask. That's up to SPIRV-Cross to figure out.

Reply:
I got tripped by the same thing: `does_shader_write_sample_mask ` isn't part of the options but is a protected member of` CompilerMSL` and is set to `true` when the SPIRV-Cross finds that the shader uses sampleMask already.

Reply:
What @Kangz said is true. We actually need to know whether the shader writes anything to the sample_mask at the moment when we run <CompilerMSL::fix_up_shader_inputs_outputs> and we can't just check if SampleMask is an active builtin then, because if we create it ourselves then we run mark_implicit_builtin for it too.